testtools/iothub_test/src/iothubtest.c

```diff
@@ -925,14 +925,15 @@ static AMQP_VALUE create_link_source(char* receive_address, filter_set filter_se
     return result;
 }
 
-static void on_message_receiver_state_changed(const void* context, MESSAGE_RECEIVER_STATE new_state, MESSAGE_RECEIVER_STATE previous_state)
+static void on_connection_state_changed(void* context, CONNECTION_STATE new_connection_state, CONNECTION_STATE previous_connection_state)
```

Comment:
It might be taxing, but what about logging both states?
That way we can use that info to debug future issues.

E.g, 
LogInfo("IoTHubTest on_connection_state_changed(%s, %s)", ENUM_TO_STRING(new_connection_state), ENUM_TO_STRING(previous_connection_state));

Reply:
CONNECTION_STATE is an enum not a set of #defines